### DISCUSSION TITLE
Getting Error: A required parameter (child) was not provided as an array received undefined in getStaticPaths for /[...child]

### DISCUSSION THREAD

**Discussion by Shivayshiva** [vercel/next.js#58387] [Help]
### Summary

Getting error while rendering SSG with dynamic parameter.

![Screenshot (3052)](https://github.com/vercel/next.js/assets/83270000/5d532192-4f0f-496e-8142-1b4ca511fc8c)
![Screenshot (3053)](https://github.com/vercel/next.js/assets/83270000/96964174-3e05-416f-b9eb-c8e4f0304b73)

**Comment by icyJoseph** [comment]
As the error is saying, your the last segment of your path is called `[...child]`, so there have to be a `child: []` parameter. Note that it has to be an array if I remember correctly.

That means that you are supporting `/pages/a/b/c/` and as many as you like, segments.

From the `getStaticPaths` signature, it looks like you only want one segment, so maybe rename the `[...child].js` path to `[postId].js`.
